### PR TITLE
[clang][compiler-rt][test] Removed dirname command substitutions from tests

### LINKED ISSUE #102384: [llvm-lit] lit internal shell failing to parse and execute command substitution syntax
lit’s internal shell cannot recognize and execute the `$()` command substitution syntax, which is used in tests across clang and compiler-rt.

Example error message:

```
FAIL: Clang :: Driver/offload-packager.c (17 of 20640)
******************** TEST 'Clang :: Driver/offload-packager.c' FAILED ********************
Exit Code: 127

Command Output (stdout):
--
# RUN: at line 28
cd $(dirname "/usr/local/google/home/harinidonthula/llvm-project/build/tools/clang/test/Driver/Output/offload-packager.c.tmp") && /usr/local/google/home/harinidonthula/llvm-project/build/bin/clang-offload-packager /usr/local/google/home/harinidonthula/llvm-project/build/tools/clang/test/Driver/Output/offload-packager.c.tmp --image=kind=openmp
# executed command: cd '$(dirname' '/usr/local/google/home/harinidonthula/llvm-project/build/tools/clang/test/Driver/Output/offload-packager.c.tmp)'
# .---command stderr------------
# | 'cd' supports only one argument
# `-----------------------------
# error: command failed with exit status: 127
```
Files with failures:
- [x] clang/test/Driver/offload-packager.c
- [x] clang/test/Modules/reduced-bmi-size.cppm
- [x] compiler-rt/test/hwasan/TestCases/hwasan_symbolize.cpp
- [ ] compiler-rt/test/msan/allocator_mapping.cpp

### PR DESCRIPTION
**Pull request by connieyzhu**
This patch rewrites tests in clang and compiler-rt that uses bash command substitution syntax $() to execute the dirname command. This is done so that the tests can be run using lit's internal shell.

Fixes https://github.com/llvm/llvm-project/issues/102384.

### CODE REVIEW

File: compiler-rt/test/hwasan/TestCases/hwasan_symbolize.cpp [hunk 1]
```diff
@@ -1,8 +1,9 @@
-// RUN: %clang_hwasan -Wl,--build-id -g %s -o %t
-// RUN: echo '[{"prefix": "'"$(realpath $(dirname %s))"'/", "link": "http://test.invalid/{file}:{line}"}]' > %t.linkify
-// RUN: %env_hwasan_opts=symbolize=0 not %run %t 2>&1 | hwasan_symbolize --html --symbols $(dirname %t) --index | FileCheck %s
-// RUN: %env_hwasan_opts=symbolize=0 not %run %t 2>&1 | hwasan_symbolize --html --linkify %t.linkify --symbols $(dirname %t) --index | FileCheck --check-prefixes=CHECK,LINKIFY %s
-// RUN: %env_hwasan_opts=symbolize=0 not %run %t 2>&1 | hwasan_symbolize --symbols $(dirname %t) --index | FileCheck %s
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: %clang_hwasan -Wl,--build-id -g %s -o %t/symbolize.exe
```
Comment (fmayer) [inline compiler-rt/test/hwasan/TestCases/hwasan_symbolize.cpp:2]: `symbolize.exe` is a bit of a confusing name. `hwasan_symbolize_test`?

Comment (connieyzhu) [reply compiler-rt/test/hwasan/TestCases/hwasan_symbolize.cpp]: I made this change in my latest commit. Thanks for the feedback!

**Review by fmayer — APPROVED** [review]
LGTM for the hwasan test.

File: clang/test/Driver/offload-packager.c [hunk 2]
```diff
@@ -3,63 +3,64 @@
...
-// RUN: diff *-amdgcn-amd-amdhsa-gfx908.2.o %t.elf.o; rm *-amdgcn-amd-amdhsa-gfx908.2.o
-// RUN: diff *-amdgcn-amd-amdhsa-gfx90a.3.o %t.elf.o; rm *-amdgcn-amd-amdhsa-gfx90a.3.o
-// RUN: not diff *-amdgcn-amd-amdhsa-gfx90c.4.o %t.elf.o
+// RUN: clang-offload-packager -o %t/package \
+// RUN:   --image=file=%t/elf.o,kind=openmp,triple=nvptx64-nvidia-cuda,arch=sm_70 \
+// RUN:   --image=file=%t/elf.o,kind=openmp,triple=nvptx64-nvidia-cuda,arch=sm_80 \
+// RUN:   --image=file=%t/elf.o,kind=openmp,triple=amdgcn-amd-amdhsa,arch=gfx908 \
+// RUN:   --image=file=%t/elf.o,kind=openmp,triple=amdgcn-amd-amdhsa,arch=gfx90a \
+// RUN:   --image=file=%t/elf.o,kind=hip,triple=amdgcn-amd-amdhsa,arch=gfx90c 
+// RUN: cd %t && clang-offload-packager %t/package --image=kind=openmp
```
Comment (ilovepi) [inline clang/test/Driver/offload-packager.c:29]: If you're going to use the path `%t/filename` you probably don't need `cd` anymore. conversely you could keep `cd` and remove the `%t` from things that used relative paths.

**Review by ilovepi — APPROVED** [review]

**Pull request merged**